guides/11_other-tutorials/create-immersive-demo.md

```diff
@@ -0,0 +1,199 @@
+
+### What you’ll build
+<gradio-app space="gradio/Gemini-Art-Critic"></gradio-app>
+
+### Prerequisites
+- Python 3.9+
+- A Gemini API key: `GEMINI_API_KEY`
+
+Install the dependencies:
+
+```bash
+pip install "fastrtc[vad, tts]" gradio google-genai python-dotenv websockets pillow
+```
+
+## 1) Encoders for audio and images
+Encoder functions to send audio as base64-encoded data and images as base64-encoded JPEG.
+
+```python
+import base64
+import numpy as np
+from io import BytesIO
+from PIL import Image
+
+def encode_audio(data: np.ndarray) -> dict:
+    """Encode audio data (int16 mono) for Gemini."""
+    return {
+        "mime_type": "audio/pcm",
+        "data": base64.b64encode(data.tobytes()).decode("UTF-8"),
+    }
+
+def encode_image(data: np.ndarray) -> dict:
+    with BytesIO() as output_bytes:
+        pil_image = Image.fromarray(data)
+        pil_image.save(output_bytes, "JPEG")
+        bytes_data = output_bytes.getvalue()
+    base64_str = str(base64.b64encode(bytes_data), "utf-8")
+    return {"mime_type": "image/jpeg", "data": base64_str}
+```
+
+
+## 2) Implement the Gemini audio-video handler
+This handler:
+- Opens a Gemini Live session on startup
+- Receives streaming audio from Gemini and yields it back to the client
+- Sends microphone audio as it arrives
+- Sends a video frame at most once per second (to avoid flooding the API)
+- Optionally sends an uploaded image (`gr.Image`) alongside the webcam frame
+
+```python
+import asyncio
+import os
+import time
+import numpy as np
+import websockets
+from dotenv import load_dotenv
+from google import genai
+from fastrtc import AsyncAudioVideoStreamHandler, wait_for_item, WebRTCError
+
+load_dotenv()
+
+class GeminiHandler(AsyncAudioVideoStreamHandler):
+    def __init__(self) -> None:
+        super().__init__(
+            "mono",
+            output_sample_rate=24000,
+            input_sample_rate=16000,
+        )
+        self.audio_queue = asyncio.Queue()
+        self.video_queue = asyncio.Queue()
+        self.session = None
+        self.last_frame_time = 0.0
+        self.quit = asyncio.Event()
+
+    def copy(self) -> "GeminiHandler":
+        return GeminiHandler()
+
+    async def start_up(self):
+        await self.wait_for_args()
+        api_key = self.latest_args[3]
+        hf_token = self.latest_args[4]
+        if hf_token is None or hf_token == "":
+            raise WebRTCError("HF Token is required")
+        os.environ["HF_TOKEN"] = hf_token
+        client = genai.Client(
+            api_key=api_key, http_options={"api_version": "v1alpha"}
+        )
+        config = {"response_modalities": ["AUDIO"], "system_instruction": "You are an art critic that will critique the artwork passed in as an image to the user. Critique the artwork in a funny and lighthearted way. Be concise and to the point. Be friendly and engaging. Be helpful and informative. Be funny and lighthearted. Be concise and to the point. Be friendly and engaging."}
```

Comment:
Repeats of `be friendly and engaging` and `be concise and to the point` not sure if intentional.